cw-orch-daemon/src/core.rs

```diff
@@ -127,6 +127,23 @@ impl DaemonAsync {
         Ok(result)
     }
 
+    /// Query a contract.
+    pub async fn query<Q: Serialize + Debug, T: Serialize + DeserializeOwned>(
```

Comment:
this is copy-pasting the `smart_query` from the AsyncWasmQuerier trait. Implement just once please